### ISSUE TITLE
Missing x3d revolution node ?

### ISSUE THREAD

**Issue by tpaviot** [andreasplesch/OCCToX3D#9]
In the latest standard revision (3.3), I can see an ```Extrusion``` node (https://www.web3d.org/documents/specifications/19775-1/V3.3/Part01/components/geometry3D.html#Extrusion). However, I can't find the ```Revolution```, that would make possible to generate a 3d geometry by rotating a face around an axis, or create a 2d geometry by rotating a curve around an axis. occt handles that type of surface:
```
enum GeomAbs_SurfaceType
{
GeomAbs_Plane,
GeomAbs_Cylinder,
GeomAbs_Cone,
GeomAbs_Sphere,
GeomAbs_Torus,
GeomAbs_BezierSurface,
GeomAbs_BSplineSurface,
GeomAbs_SurfaceOfRevolution,
GeomAbs_SurfaceOfExtrusion,
GeomAbs_OffsetSurface,
GeomAbs_OtherSurface
};
```

Is there any on going work about that (may be the 4.0 version) ?

**Comment by andreasplesch** [comment]
Yes, there is no Revolution node in X3D. Generally, the tendency has been in X3D to rely more on the modelers to provide tessellated geometry, and focus more on event processing and dynamic aspects of a scene.

x3dom has a few more, non-standard geometric primitives.

This includes SolidOfRevolution:
https://doc.x3dom.org/author/Geometry3DExt/SolidOfRevolution.html

For plane, there is also non-standard Plane node:
https://doc.x3dom.org/author/Geometry3D/Plane.html
For standard X3D there is 2DRectangle which can be transformed:
https://doc.x3dom.org/author/Geometry2D/Rectangle2D.html

For Torus, there is a non-standard Torus node:
https://doc.x3dom.org/author/Geometry3D/Torus.html

There is no OffsetSurface equivalent.
